src/components/Netcat/Netcat.tsx

```diff
@@ -30,19 +21,46 @@ interface FormValuesType {
     filePath: string;
 }
 
-//Netcat Options
+//Netcat Options Configuration
 const netcatOptions = ["Listen", "Connect", "Port Scan", "Send File", "Receive File", "Website Port Scan"];
 
+/**
+ * The Netcat component.
+ * @returns The Netcat component.
+ */
 //Tool name must be capital or jsx will cry out errors :P
 const NetcatTool = () => {
-    var [output, setOutput] = useState("");
+    //Component State Variables
+    var [output, setOutput] = useState(""); // State variable to store the output of the command execution.
     const [selectedScanOption, setSelectedNetcatOption] = useState("");
-    const [pid, setPid] = useState("");
-    const [loading, setLoading] = useState(false);
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
-    const [checkedVerboseMode, setCheckedVerboseMode] = useState(false);
-
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved
+    const [checkedVerboseMode, setCheckedVerboseMode] = useState(false); // State variable to indicate whether the verbose mode is enabled.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    // Component Constants.
+    const title = "Netcat Tool"; // Title of the component.
+    const description =
+        "Netcat is a powerful tool that is used to connect two machines together for communication and for other uses."; // Description of the component.
+    const steps =
+        "How to use this netcat tool:\n" +
+        "- If you want to listen for connections for chat or reverse shell choose the Interactive shell/Listen option and provide \n a port number.\n" +
```

Comment:
The format for this should be Step 1: blah, Step 2: blah. Refer to a tool like WhatWeb for an example.

Reply:
I'm still thinking about the word "Option" for the description of this tool as for this tool, you just need to choose between the options and not really follow any steps to start.

Reply:
I changed to Step and update the description a little